rs/bitcoin/kyt/tests/tests.rs

```diff
@@ -110,64 +182,135 @@ fn test_get_inputs() {
-\xc8\x47\x25\x2b\xfc\xf4\x24\xb3\x75\x8f\xd1\x57\x92\xef\xf4\xa4\xaa\x01\x21\x02\xc4\xc6\x9e\x4d\
-\x36\x4b\x3e\xdf\x84\xb5\x20\xa0\x18\xd5\x7e\x71\xfa\xce\x19\x7e\xc8\xf9\x46\x43\x60\x7e\x4a\xca\
-\x70\xdc\x82\xc1\xfd\xff\xff\xff\x02\x10\x27\x00\x00\x00\x00\x00\x00\x19\x76\xa9\x14\x62\xe9\x07\
-\xb1\x5c\xbf\x27\xd5\x42\x53\x99\xeb\xf6\xf0\xfb\x50\xeb\xb8\x8f\x18\x88\xac\x00\x96\x00\x00\x00\
-\x00\x00\x00\x19\x76\xa9\x14\xb9\x73\x68\xd8\xbf\x0a\x37\x69\x00\x85\x16\x57\xf3\x7f\xbe\x73\xa6\
-\x56\x61\x33\x88\xac\xb3\xa3\x0c\x00"
-                .to_vec(),
-        }),
-    });
+            subnet_id: canister_http_requests[0].subnet_id,
+            request_id: canister_http_requests[0].request_id,
+            response: CanisterHttpResponse::CanisterHttpReply(CanisterHttpReply {
+                status: 200,
+                headers: vec![],
+                body: b"\
+    \x02\x00\x00\x00\x01\x82\xc8\x5d\xe7\x4d\x19\xbb\x36\x16\x2f\xca\xef\xc7\xe7\x70\x15\x65\xb0\x2d\
+    \xf6\x06\x0f\x8e\xcf\x49\x64\x63\x37\xfc\xe8\x59\x37\x07\x00\x00\x00\x6a\x47\x30\x44\x02\x20\x15\
+    \xf2\xc7\x7a\x3b\x95\x13\x73\x7a\xa2\x86\xb3\xe6\x06\xf9\xb6\x82\x1c\x6d\x5d\x35\xe5\xa9\x58\xe0\
+    \x1f\x65\x76\xec\xdf\xac\x76\x02\x20\x4e\xad\x06\x1d\xe8\x3c\x5b\x07\x25\x8e\xfd\x2f\x44\x3d\xeb\
+    \xc8\x47\x25\x2b\xfc\xf4\x24\xb3\x75\x8f\xd1\x57\x92\xef\xf4\xa4\xaa\x01\x21\x02\xc4\xc6\x9e\x4d\
+    \x36\x4b\x3e\xdf\x84\xb5\x20\xa0\x18\xd5\x7e\x71\xfa\xce\x19\x7e\xc8\xf9\x46\x43\x60\x7e\x4a\xca\
+    \x70\xdc\x82\xc1\xfd\xff\xff\xff\x02\x10\x27\x00\x00\x00\x00\x00\x00\x19\x76\xa9\x14\x62\xe9\x07\
+    \xb1\x5c\xbf\x27\xd5\x42\x53\x99\xeb\xf6\xf0\xfb\x50\xeb\xb8\x8f\x18\x88\xac\x00\x96\x00\x00\x00\
+    \x00\x00\x00\x19\x76\xa9\x14\xb9\x73\x68\xd8\xbf\x0a\x37\x69\x00\x85\x16\x57\xf3\x7f\xbe\x73\xa6\
+    \x56\x61\x33\x88\xac\xb3\xa3\x0c\x00"
+                    .to_vec(),
+            }),
+        });
 
     let result = env
         .await_call(call_id)
         .expect("the fetch request didn't finish");
 
     match &result {
         WasmResult::Reply(bytes) => {
-            let response = Decode!(bytes, Vec<String>).unwrap();
-            assert_eq!(
-                response,
-                vec!["1HuaGHYa4rjqpcqkm1thE48s88rf1QpR5J".to_string()],
-            );
+            let response = Decode!(bytes, CheckTransactionResponse).unwrap();
+            assert!(matches!(response, CheckTransactionResponse::Passed));
         }
         WasmResult::Reject(msg) => panic!("unexpected reject: {}", msg),
     }
+
+    let cycles_after = env.cycle_balance(setup.universal_canister);
+    let expected_cost =
+        CHECK_TRANSACTION_CYCLES_SERVICE_FEE + 2 * get_tx_cycle_cost(INITIAL_BUFFER_SIZE);
+    let actual_cost = cycles_before - cycles_after;
+    // Since we use universal_canister, actual_cost is greater than expected_cost
+    assert!(actual_cost > expected_cost);
+    assert!(actual_cost - expected_cost < 7_000_000);
```

Comment:
Where does `7_000_000` come from?

Reply:
The cycle difference before/after is computed by measuring the balance of the universal canister, because only a canister can pay for cycles. But then this universal canister also burns cycles by proxying this call (e.g. some decoding work in there). I'll add some explanation in comments.

Reply:
More comments are added to explain this.